halsim_gui/_ext/main.cpp

```diff
@@ -0,0 +1,21 @@
+
+#include <robotpy_build.h>
+
+// these aren't exported in an accessible header, so define them here
+namespace wpi::gui {
+void AddLateExecute(std::function<void()> execute);
+} // namespace wpi::gui
+
+RPYBUILD_PYBIND11_MODULE(m) {
+
+  m.def("_kill_on_signal", []() {
+    wpi::gui::AddLateExecute([] {
+      {
+        py::gil_scoped_acquire gil;
+        if (PyErr_CheckSignals() == -1) {
+          throw py::error_already_set();
```

Comment:
@PeterJohnson if I just throw an exception on Windows/OSX from here... anything bad going to happen? Seems to be fine on Linux. 

Eventually pybind11 will catch the exception and propagate it as a python exception to whomever called `HAL_RunMain`.

Reply:
If I just call `wpi::gui::Exit`, the python error indicator is still set, and python becomes very unhappy when you pop back into python code with an active exception.